Source/sources/Castro_sources.cpp

```diff
@@ -14,8 +14,6 @@ Castro::apply_source_to_state(MultiFab& state, MultiFab& source, Real dt, int ng
     AMREX_ASSERT(state.nGrow() >= ng);
 
     MultiFab::Saxpy(state, dt, source, 0, 0, NUM_STATE, ng);
-
-    clean_state(state);
```

Comment:
Since it looks like every call to apply_source_to_state still gets an explicit clean_state call after it, can we put the clean_state call back here and simplify the code a little?

Reply:
the reason I took it out is because eventually I want to get rid of the `clean_state()` that takes a MF in favor of the version that works directly on state date.  I got rid of all but one of those, but hope to remove the last in the future.

Reply:
But can we put it back in, just using the new interface?

Reply:
there was a reason why I didn't think so, but let me look again...

Reply:
oh... so I'd need to change the interface of the source stuff to pass in an `is_new` like variable, I guess that's doable.

Reply:
okay... I remember now... I need to know which bin it is in the StateData for the 4th order stuff, hence the change of the interface.  But if I pass that info into apply_source_to_state, we are fine.  So I'll make that change